app/components/forms/PlannerForm.tsx

```diff
@@ -0,0 +1,191 @@
+        <p className="text-sm">
+          What is your most important goal for this week?
+        </p>
+        <input
+          type="text"
+          value={bigGoal}
+          onChange={(e) => setBigGoal(e.target.value)}
+          className="w-full p-2 border rounded"
+          placeholder="Describe your most important goal for this week..."
+        />
+        <div className="grid grid-cols-1 lg:grid-cols-2 gap-4 pt-4">
+          {Object.entries(sections).map(([section, items]) => (
+            <div key={section} className="space-y-2">
+              <h2 className="font-semibold">
+                This week I need to{' '}
+                <span>{section.replace(/([A-Z])/g, ' $1')}</span> ...
+              </h2>
+              {items.map((item, index) => (
+                <div
+                  key={index}
+                  className="flex gap-2 items-center bg-primary p-2 rounded shadow-sm"
+                >
+                  <input
+                    type="text"
+                    value={item.text}
+                    onChange={(e) =>
+                      handleChange(
+                        section as SectionKeys,
+                        index,
+                        'text',
+                        e.target.value
+                      )
+                    }
+                    className="p-2 border rounded w-full"
+                    placeholder="Task"
+                  />
+                  <input
+                    type="date"
+                    value={item.date}
+                    onChange={(e) =>
+                      handleChange(
+                        section as SectionKeys,
+                        index,
+                        'date',
+                        e.target.value
+                      )
+                    }
+                    className="p-2 border rounded w-1/4"
+                  />
+                  <TagSelector
+                    selectedTag={item.tag || ''}
+                    onTagSelect={(tag) =>
+                      handleChange(section as SectionKeys, index, 'tag', tag)
+                    }
+                  />
+                  <button
+                    type="button"
+                    onClick={() =>
+                      togglePriority(section as SectionKeys, index)
+                    }
+                    className={`p-2 rounded ${
+                      priorities.includes(item)
+                        ? 'bg-yellow-400'
+                        : 'bg-gray-200'
+                    }`}
+                  >
+                    ⭐
+                  </button>
+                </div>
+              ))}
+              <button
+                type="button"
+                onClick={() => addField(section as SectionKeys)}
+                className="text-blue-500 mt-1"
+              >
+                + Add More
+              </button>
+            </div>
+          ))}
+        </div>
```

Comment:
I like this whole input, and it could work better than the current input we use in AddTasks. 

We should separate it out and use it across the app.